packages/ui/src/components/va-data-table/hooks/useAnimationName.ts

```diff
@@ -1,31 +1,38 @@
-
 import { computed, Ref, ref, watch } from 'vue'
+
 import type { DataTableRow } from '../types'
 
-interface useSelectableProps {
+type AnimationName = 'shuffle' | 'fade'
+
+interface useAnimationNameProps {
   currentPage: number | undefined
   animated: boolean
 }
 
-export default function usePaginatedRows (
-  props: useSelectableProps,
+export default function useAnimationName (
+  props: useAnimationNameProps,
   rows: Ref<DataTableRow[]>,
 ) {
-  const animationName = ref('table-transition-shuffle')
+  const animationType = ref<AnimationName>('shuffle')
+
+  const animationName = computed(() => props.animated ? `table-transition-${animationType.value}` : '')
+
   const oldRowsLength = ref(rows.value.length)
   const isDifferentRowLength = computed(() => rows.value.length !== oldRowsLength.value)
 
-  watch(rows, (newRows) => {
-    const animationType = isDifferentRowLength.value || newRows.length > 50 ? 'fade' : 'shuffle'
+  watch(rows, (newRows, oldRows) => {
+    const hasRows = !!(newRows.length && oldRows.length)
 
-    animationName.value = props.animated ? `table-transition-${animationType}` : ''
+    animationType.value = newRows.length > 50 || (isDifferentRowLength.value && hasRows)
+      ? 'fade'
+      : 'shuffle'
 
     oldRowsLength.value = newRows.length
   })
 
   watch(() => props.currentPage, () => {
     if (!isDifferentRowLength.value) {
-      animationName.value = props.animated ? 'table-transition-shuffle' : ''
+      animationType.value = 'shuffle'
```

Comment:
The `animated` prop works because it is used in the `animationName` variable:
```ts
const animationName = computed(() => props.animated ? `table-transition-${animationType.value}` : '')
```